boot/seal.go

```diff
@@ -270,6 +270,81 @@ type ResealKeyForBootChainsParams struct {
 	RoleToBlName map[bootloader.Role]string
 }
 
+// BootChains constructs the boot chains which may be observed when booting the
+// device such that they can be used as an input for resealing of encryption
+// keys.
+func BootChains(modeenv *Modeenv) (*ResealKeyForBootChainsParams, error) {
```

Comment:
I thought the idea we discussed was to have WithBootChains ? did you go this way because is a bit easier to explain? do we have other use cases for WithModeenv?

Reply:
Fair point, perhaps there is no need to provide a helper for touching modeenv. I've pushed a patch.